### PR TITLE
service: fix resolver to always return a username in Resolve2

### PR DESCRIPTION
**Pull request by maxtaco**
- still not required in Resolve(), but we can change that.
- add RPC-tests to make sure that we don't get messed up w/r/t cross-caching
  with the two RPCs.

### CODE REVIEW

**Comment by patrickxb** [comment]
LGTM

File: go/systests/rpc_test.go [hunk 1]
```diff
@@ -57,6 +57,19 @@ func testIdentifyResolve2(t *testing.T, g *libkb.GlobalContext) {
...
 
+	if _, err := cli.Resolve(context.TODO(), "uid:eb72f49f2dde6429e5d78003dae0c919"); err != nil {
+		t.Fatalf("Resovle failed: %v\n", err)
+	}
+
+	// We don't want to hit the cache, since the previous lookup never hit the
+	// server.  For Resolve2, we have to, since we need a username.  So test that
+	// here.
+	if res, err := cli.Resolve2(context.TODO(), "uid:eb72f49f2dde6429e5d78003dae0c919"); err != nil {
+		t.Fatalf("Resovle failed: %v\n", err)
```
Comment (jzila) [inline go/systests/rpc_test.go]: Typo in string: s/Resovle/Resolve

File: go/systests/rpc_test.go [hunk 2]
```diff
@@ -57,6 +57,19 @@ func testIdentifyResolve2(t *testing.T, g *libkb.GlobalContext) {
 		t.Fatalf("failed to get new identifyclient: %v", err)
 	}
 
+	if _, err := cli.Resolve(context.TODO(), "uid:eb72f49f2dde6429e5d78003dae0c919"); err != nil {
+		t.Fatalf("Resovle failed: %v\n", err)
```
Comment (jzila) [inline go/systests/rpc_test.go]: Typo in string: s/Resovle/Resolve

**Pull request merged**